### ISSUE TITLE
Documents are not removed if page is hidden

### ISSUE THREAD

**Issue by sascha-egerer** [TYPO3-Solr/ext-solr#258]
If i hide a page the records stored on that page are not removed from the solr index. So records of hidden pages are still shown in the search result.

**Comment by irnnr** [comment]
The record monitor only checks changed records ATM. To fix this we would have to check whether the changed record is a page and then iterate over the records it contains. 

However, AFAIR there's no reverse relation, so we would have to check all (configured) tables for records with the page as their parent page ID.

**Comment by sascha-egerer** [comment]
> However, AFAIR there's no reverse relation, so we would have to check all (configured) tables for records with the page as their parent page ID. 

Thats correct. There is currently no way to do that so i used the easiest way until this can be implemented with better core support

**Comment by thomashohn** [comment]
Whats the status on this bug?

**Comment by npoggenburg** [comment]
Hey,
this is still a thing. If i delete/hide a news record and start the scheduler again (its still at 100% indexed) deleted/hidden records do not get checked again and are not removed from index. 
Any ideas on this topic?

**Comment by thomashohn** [comment]
I extended the garbage collector to support this - even though this is not a general applicable solution it works fine for the sites I index..
